components/channel_view/channel_view.jsx

```diff
@@ -43,18 +43,31 @@ export default class ChannelView extends React.PureComponent {
     }
 
     static getDerivedStateFromProps(props, state) {
-        let updatedState = {url: props.match.url};
-        if (!state.url || props.match.url !== state.url) {
-            updatedState = {...updatedState, deferredPostView: ChannelView.createDeferredPostView()};
+        let updatedState = {};
+        if (props.match.url !== state.url) {
+            updatedState = {deferredPostView: ChannelView.createDeferredPostView(), url: props.match.url};
         }
 
-        return updatedState;
+        if (props.channelId !== state.channelId) {
+            updatedState = {...updatedState, channelId: props.channelId, prevChannelId: state.channelId};
+        }
+
+        if (Object.keys(updatedState).length) {
+            return updatedState;
```

Comment:
Have we tested to see if there's a difference between returning an empty object and null from `getDerivedStateFromProps`? I assume React does a check like this internally